backend/spellchecker/proto/spellchecker.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+option java_package = "com.papeeria";
+
+service Spellcheck {
+    rpc CheckText(Text) returns (Suggestions) {}
+}
+
+message Text {
+    bytes text = 1;
+    repeated string language = 2;
+}
+
+message Suggestions {
+    string json = 1;
```

Comment:
one of the purposes of protocol buffers is to provide typed interface for data exchange between servers. Please replace this json with typed interface, like `map<string, Replacements> suggestions`, use it in the spellchecker and remove your own json-serialization